### ISSUE TITLE
alpha.13 throws LCB_ERR_SDK_FEATURE_UNAVAILABLE (1029)

### ISSUE THREAD

**Issue by bwgjoseph** [bwgjoseph/mongoose-vs-ottoman#39]
Upon upgrading to `alpha.13`, this error throws immediately. `alpha.13` is pretty much unusable for me, is there any configuration change I should know about?

```js
$ npm test

> mongoose-vs-ottoman@1.0.0 test Z:\mongoose-vs-ottoman
> mocha --require ts-node/register test/ts/*.ts --exit



  test create function
    √ mongoose - should create new doc (71ms)
    1) ottoman - should create new doc


  1 passing (205ms)
  1 failing

  1) test create function
       ottoman - should create new doc:
     Error: LCB_ERR_SDK_FEATURE_UNAVAILABLE (1029): The requested feature is not supported by the client, either because of settings in the configured instance, or because of options disabled at the time the library was compiled
      at CbConnection.<anonymous> (node_modules\couchbase\lib\connection.js:284:38)
      at Connection._maybeBFwd (node_modules\couchbase\lib\connection.js:241:12)
      at Connection.store (node_modules\couchbase\lib\connection.js:284:10)
      at Z:\mongoose-vs-ottoman\node_modules\couchbase\lib\collection.js:558:18
      at Z:\mongoose-vs-ottoman\node_modules\couchbase\lib\promisehelper.js:30:7
      at new Promise (<anonymous>)
      at Function.wrap (node_modules\couchbase\lib\promisehelper.js:29:12)
      at Collection._store (node_modules\couchbase\lib\collection.js:557:26)
      at Collection.insert (node_modules\couchbase\lib\collection.js:616:17)
      at Object.store (node_modules\ottoman\src\handler\store.ts:16:31)
      at Z:\mongoose-vs-ottoman\node_modules\ottoman\srcmodel\utils\store-life-cycle.ts:31:24
      at Generator.next (<anonymous>)
      at fulfilled (node_modules\ottoman\libmodel\utils\store-life-cycle.js:5:58)



npm ERR! Test failed.  See above for more details.
```

I am still running on `6.6.0`. I tested on a few different test cases, and all returns this error.

**Comment by josephstgh** [comment]
I ran on another machine, cleared away existed `node_modules` and `package-lock.json` and ran `npm i` to re-generate/install again

In the other machine, I got this error

```js
C:\mongoose-vs-ottoman>npm test

> mongoose-vs-ottoman@1.0.0 test C:\mongoose-vs-ottoman
> mocha --require ts-node/register test/ts/*.ts --exit



  1) "before all" hook in "{root}"

  0 passing (136ms)
  1 failing

  1) "before all" hook in "{root}":
     Error: path exists
      at _getWrappedErr (node_modules\couchbase\lib\errors.js:904:14)
      at Object.wrapLcbErr (node_modules\couchbase\lib\errors.js:1009:20)
      at C:\mongoose-vs-ottoman\node_modules\couchbase\lib\queryexecutor.js:126:26



npm ERR! Test failed.  See above for more details.
```

**Comment by AV25242** [comment]
@bwgjoseph  one basic thing looking at this error **"The requested feature is not supported by the client, either because of settings in the configured instance, or because of options disabled at the time the library was compiled"**
is that you will have to setup collectionName=_default in the global configuration (you will have to do that for a server instance in your case 6.6 that doesn't have scopes or collections enabled)

**Comment by bwgjoseph** [comment]
@AV25242 yes. I have that declared in my initialization. 

The path exist error, I'm facing on 2 different machine.

**Comment by AV25242** [comment]
can you please share your repository ?

**Comment by AV25242** [comment]
@bwgjoseph  looks like there are 2 different problems

a) The path exist error
b) SchemaType (nested) not working 

Both of them works for our developer on local and integration machines. 

They wanted to check your code, can you please share the repository and environment information like

CB Server, Machine you are running on etc

@httpJunkie  for your reference

**Comment by josephstgh** [comment]
The repository is this current repo. 

The current code in the repo does not include the nested schema declaration.

**Comment by AV25242** [comment]
Thanks @bwgjoseph will have someone take a look

**Comment by AV25242** [comment]
@bwgjoseph  can you check this response from our developer ?


a) The path exist error
Should be something wrong here in line 10 file test/ts/setup/util.ts inside the function removeDocuments

I think the bucket name is wrong because of this, the "path exists" error is throwing

**Comment by josephstgh** [comment]
The code for this has been since 2 months ago, and it was all along working until upgraded to alpha.13 and then downgrade back to alpha.12.

I just tested to remove `removeDocuments` method, it throws me

`Error: LCB_ERR_SDK_FEATURE_UNAVAILABLE (1029): The requested feature is not supported by the client, either because of settings in the configured instance, or because of options disabled at the time the library was compiled`

**Comment by josephstgh** [comment]
@AV25242 @httpJunkie 

I have tried to upgrade to 7.0.0-beta and tested with the following scenario

1. 7.0.0-beta, **alpha.12**, with removeDocument method ==> `path exist error`
2. 7.0.0-beta, **alpha.12**, without removeDocument method ==> `path exist error`
3. 7.0.0-beta, **alpha.13**, with removeDocument method ==> `path exist error`
4. 7.0.0-beta, **alpha.13**, without removeDocument method ==> `path exist error`

In all the 4 test above, I'm only running `ottoman - should create new doc` in `create.test.ts` test case

It seem like with `7.0.0-beta`, I don't see the `LCB_ERR_SDK_FEATURE_UNAVAILABLE` error but `path exist error` remains

The UI

![image](https://user-images.githubusercontent.com/47846485/105128462-0cfd3780-5b1e-11eb-82f7-4bc2888f194c.png)

The dependency tree of `alpha.13`

```js
+-- ottoman@2.0.0-alpha.13
| +-- couchbase@3.1.1
| | +-- bindings@1.5.0
| | | `-- file-uri-to-path@1.0.0
| | +-- debug@4.3.1
| | | `-- ms@2.1.2 deduped
| | +-- nan@2.14.2
| | +-- parse-duration@0.4.4
| | +-- prebuild-install@6.0.0
| | | +-- detect-libc@1.0.3
| | | +-- expand-template@2.0.3
| | | +-- github-from-package@0.0.0
| | | +-- minimist@1.2.5
| | | +-- mkdirp-classic@0.5.3
| | | +-- napi-build-utils@1.0.2
| | | +-- node-abi@2.19.3
| | | | `-- semver@5.7.1 deduped
| | | +-- noop-logger@0.1.1
| | | +-- npmlog@4.1.2
| | | | +-- are-we-there-yet@1.1.5
| | | | | +-- delegates@1.0.0
| | | | | `-- readable-stream@2.3.7 deduped
| | | | +-- console-control-strings@1.1.0
| | | | +-- gauge@2.7.4
| | | | | +-- aproba@1.2.0
| | | | | +-- console-control-strings@1.1.0 deduped
| | | | | +-- has-unicode@2.0.1
| | | | | +-- object-assign@4.1.1
| | | | | +-- signal-exit@3.0.3
| | | | | +-- string-width@1.0.2 deduped
| | | | | +-- strip-ansi@3.0.1 deduped
| | | | | `-- wide-align@1.1.3 deduped
| | | | `-- set-blocking@2.0.0 deduped
| | | +-- pump@3.0.0
| | | | +-- end-of-stream@1.4.4
| | | | | `-- once@1.4.0 deduped
| | | | `-- once@1.4.0 deduped
| | | +-- rc@1.2.8
| | | | +-- deep-extend@0.6.0
| | | | +-- ini@1.3.8
| | | | +-- minimist@1.2.5 deduped
| | | | `-- strip-json-comments@2.0.1
| | | +-- simple-get@3.1.0
| | | | +-- decompress-response@4.2.1
| | | | | `-- mimic-response@2.1.0
| | | | +-- once@1.4.0 deduped
| | | | `-- simple-concat@1.0.1
| | | +-- tar-fs@2.1.1
| | | | +-- chownr@1.1.4
| | | | +-- mkdirp-classic@0.5.3 deduped
| | | | +-- pump@3.0.0 deduped
| | | | `-- tar-stream@2.2.0
| | | |   +-- bl@4.0.3
| | | |   | +-- buffer@5.7.1
| | | |   | | +-- base64-js@1.5.1
| | | |   | | `-- ieee754@1.2.1
| | | |   | +-- inherits@2.0.4 deduped
| | | |   | `-- readable-stream@3.6.0 deduped
| | | |   +-- end-of-stream@1.4.4 deduped
| | | |   +-- fs-constants@1.0.0
| | | |   +-- inherits@2.0.4 deduped
| | | |   `-- readable-stream@3.6.0
| | | |     +-- inherits@2.0.4 deduped
| | | |     +-- string_decoder@1.1.1 deduped
| | | |     `-- util-deprecate@1.0.2 deduped
| | | +-- tunnel-agent@0.6.0
| | | | `-- safe-buffer@5.2.1 deduped
| | | `-- which-pm-runs@1.0.0
| | `-- qs@6.9.6
| +-- jsonpath@1.1.0
| | +-- esprima@1.2.2
| | +-- static-eval@2.0.2
| | | `-- escodegen@1.14.3
| | |   +-- esprima@4.0.1
| | |   +-- estraverse@4.3.0
| | |   +-- esutils@2.0.3
| | |   +-- optionator@0.8.3
| | |   | +-- deep-is@0.1.3
| | |   | +-- fast-levenshtein@2.0.6
| | |   | +-- levn@0.3.0
| | |   | | +-- prelude-ls@1.1.2 deduped
| | |   | | `-- type-check@0.3.2 deduped
| | |   | +-- prelude-ls@1.1.2
| | |   | +-- type-check@0.3.2
| | |   | | `-- prelude-ls@1.1.2 deduped
| | |   | `-- word-wrap@1.2.3
| | |   `-- source-map@0.6.1 deduped
| | `-- underscore@1.7.0
| `-- uuid@8.3.2
```

The dependency tree of `alpha.12`

```js
+-- ottoman@2.0.0-alpha.12
| +-- couchbase@3.1.1
| | +-- bindings@1.5.0
| | | `-- file-uri-to-path@1.0.0
| | +-- debug@4.3.1
| | | `-- ms@2.1.2 deduped
| | +-- nan@2.14.2
| | +-- parse-duration@0.4.4
| | +-- prebuild-install@6.0.0
| | | +-- detect-libc@1.0.3
| | | +-- expand-template@2.0.3
| | | +-- github-from-package@0.0.0
| | | +-- minimist@1.2.5
| | | +-- mkdirp-classic@0.5.3
| | | +-- napi-build-utils@1.0.2
| | | +-- node-abi@2.19.3
| | | | `-- semver@5.7.1 deduped
| | | +-- noop-logger@0.1.1
| | | +-- npmlog@4.1.2
| | | | +-- are-we-there-yet@1.1.5
| | | | | +-- delegates@1.0.0
| | | | | `-- readable-stream@2.3.7 deduped
| | | | +-- console-control-strings@1.1.0
| | | | +-- gauge@2.7.4
| | | | | +-- aproba@1.2.0
| | | | | +-- console-control-strings@1.1.0 deduped
| | | | | +-- has-unicode@2.0.1
| | | | | +-- object-assign@4.1.1
| | | | | +-- signal-exit@3.0.3
| | | | | +-- string-width@1.0.2 deduped
| | | | | +-- strip-ansi@3.0.1 deduped
| | | | | `-- wide-align@1.1.3 deduped
| | | | `-- set-blocking@2.0.0 deduped
| | | +-- pump@3.0.0
| | | | +-- end-of-stream@1.4.4
| | | | | `-- once@1.4.0 deduped
| | | | `-- once@1.4.0 deduped
| | | +-- rc@1.2.8
| | | | +-- deep-extend@0.6.0
| | | | +-- ini@1.3.8
| | | | +-- minimist@1.2.5 deduped
| | | | `-- strip-json-comments@2.0.1
| | | +-- simple-get@3.1.0
| | | | +-- decompress-response@4.2.1
| | | | | `-- mimic-response@2.1.0
| | | | +-- once@1.4.0 deduped
| | | | `-- simple-concat@1.0.1
| | | +-- tar-fs@2.1.1
| | | | +-- chownr@1.1.4
| | | | +-- mkdirp-classic@0.5.3 deduped
| | | | +-- pump@3.0.0 deduped
| | | | `-- tar-stream@2.2.0
| | | |   +-- bl@4.0.3
| | | |   | +-- buffer@5.7.1
| | | |   | | +-- base64-js@1.5.1
| | | |   | | `-- ieee754@1.2.1
| | | |   | +-- inherits@2.0.4 deduped
| | | |   | `-- readable-stream@3.6.0 deduped
| | | |   +-- end-of-stream@1.4.4 deduped
| | | |   +-- fs-constants@1.0.0
| | | |   +-- inherits@2.0.4 deduped
| | | |   `-- readable-stream@3.6.0
| | | |     +-- inherits@2.0.4 deduped
| | | |     +-- string_decoder@1.1.1 deduped
| | | |     `-- util-deprecate@1.0.2 deduped
| | | +-- tunnel-agent@0.6.0
| | | | `-- safe-buffer@5.2.1 deduped
| | | `-- which-pm-runs@1.0.0
| | `-- qs@6.9.6
| +-- jsonpath@1.1.0
| | +-- esprima@1.2.2
| | +-- static-eval@2.0.2
| | | `-- escodegen@1.14.3
| | |   +-- esprima@4.0.1
| | |   +-- estraverse@4.3.0
| | |   +-- esutils@2.0.3
| | |   +-- optionator@0.8.3
| | |   | +-- deep-is@0.1.3
| | |   | +-- fast-levenshtein@2.0.6
| | |   | +-- levn@0.3.0
| | |   | | +-- prelude-ls@1.1.2 deduped
| | |   | | `-- type-check@0.3.2 deduped
| | |   | +-- prelude-ls@1.1.2
| | |   | +-- type-check@0.3.2
| | |   | | `-- prelude-ls@1.1.2 deduped
| | |   | `-- word-wrap@1.2.3
| | |   `-- source-map@0.6.1 deduped
| | `-- underscore@1.7.0
| `-- uuid@8.3.2
```

I did a delta check against the 2 tree, the only difference is `alpha.12` to `alpha.13`

Also, between upgrading from 12 to 13, I did remove `node_modules` and `package-lock` before running `npm i`

**Comment by josephstgh** [comment]
Based on the earlier conversation, the current workaround is to install `couchbase@3.1.0` and create the index `CREATE PRIMARY INDEX 'ottoman-test-index' ON 'testBucket'` with `Couchbase 7.0.0-enterprise-beta`

Will keep this open first, waiting for the next alpha release

**Comment by AV25242** [comment]
Alpha.14 is released

FIX: update ottoman to work with couchbase 3.1.1 #368
CHANGE FEAT: rename the replace method to replaceById  #365
CHANGE FEAT: update bulk operation response errors  #363
